File: week6-python/pset6/cash.py
from cs50 import get_float

def get_cents():
    while True:
        dollars = get_float("Change: ")
        if (dollars > 0):
            return dollars
            break


def calculate_quarters(dollars):
    quarter = 0
    while (dollars >= 25):
        dollars = dollars - 25
        quarter += 1
    return quarter


def calculate_dimes(dollars):
    dime = 0
    while (dollars >= 10):
        dollars = dollars - 10
        dime += 1
    return dime


def calculate_nickels(dollars):
    nickel = 0
    while (dollars >= 5):
        dollars = dollars - 5
        nickel += 1
    return nickel


def calculate_pennies(dollars):
    penny = 0
    while (dollars >= 1):
        dollars = dollars - 1
        penny += 1
    return penny



def main():
    dollars = get_cents()
    dollars *= 100

    quarters = calculate_quarters(dollars)
    dollars = dollars - quarters * 25

    dimes = calculate_dimes(dollars)
    dollars = dollars - dimes * 10

    nickels = calculate_nickels(dollars)
    dollars = dollars - nickels * 5

    pennies = calculate_pennies(dollars)
    dollars = dollars - pennies * 1

    coins = quarters + dimes + nickels + pennies

    print(f"{coins}")


main()

File: week1-C/pset1/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    int cents;
    do{
        cents = get_int("How much do I owe you? ");
    }while(cents < 0);

    return cents;
}

int calculate_quarters(int cents){
    int quarter = 0;
    while(cents >= 25){
        cents = cents - 25;
        quarter++;
    }
    return quarter;
}


int calculate_dimes(int cents)
{
    int dime = 0;
    while(cents >= 10){
        cents = cents - 10;
        dime++;
    }
    return dime;
}

int calculate_nickels(int cents)
{
    int nickel = 0;
    while(cents >= 5){
        cents = cents - 5;
        nickel++;
    }
    return nickel;
}

int calculate_pennies(int cents)
{
    int penny = 0;
    while(cents >=1){
        cents = cents - 1;
        penny++;
    }
    return penny;
}
